scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/syntax/SCollectionSyntax.scala

```diff
@@ -84,6 +88,22 @@ final class SCollectionTableRowOps[T <: TableRow](private val self: SCollection[
     val param = TableRowJsonWriteParam(numShards, compression)
     self.covary[TableRow].write(TableRowJsonIO(path))(param)
   }
+
+  /**
+   * Save this SCollection to partitioned BigQuery tables using the specified table function.
+   * Note that elements must be of type
+   * [[com.google.api.services.bigquery.model.TableRow TableRow]].
+   */
+  def saveAsPartitionedTable(
```

Comment:
I think we could just use the same naming scheme here and have a `saveAsBigQueryTable` with the curried fn.

We probably could have one method that accepts `DynamicDestination`

We also would want to take into account the `Format[T]` because we will want this to work with `GenericRecord`.